src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -352,24 +355,64 @@ FixedwingPositionControl::get_demanded_airspeed()
 	return altctrl_airspeed;
 }
 
+
 float
-FixedwingPositionControl::calculate_target_airspeed(float airspeed_demand, const Vector2f &ground_speed)
+FixedwingPositionControl::get_cruise_airspeed_setpoint(const hrt_abstime &now, const float pos_sp_cru_airspeed,
+		const Vector2f &ground_speed, float dt)
 {
+	float airspeed_setpoint = _param_fw_airspd_trim.get();
+
+	if (_speed_mode_current == FW_SPEED_MODE::FW_SPEED_MODE_ECO) {
```

Comment:
Rather than using the global variable, can we make this a parameter of the function? That way the behavior is more consistent

Reply:
Yep, have a look at https://github.com/PX4/PX4-Autopilot/pull/18605/commits/008a14ddf5a063c538de6a95d06fc562fbe39222. Do you like that more? Agree that it's more consistent with the current architecture of FW Pos Control